Src/HALALMock/Services/DigitalOutputService/DigitalOutputService.cpp

```diff
@@ -54,5 +57,5 @@ void DigitalOutputService::toggle(uint8_t id){
 	}
 
 	Pin pin = DigitalOutputService::service_ids[id];
-	HAL_GPIO_TogglePin(pin.port, pin.gpio_pin);
+	(*gpio_memory + pin_offset[pin.gpio_pin]) ^= 0x01;
```

Comment:
```suggestion
	(*gpio_memory + pin_offset[pin.gpio_pin]) = (*gpio_memory + pin_offset[pin.gpio_pin]) == PinState::HIGH ? PinState::LOW : PinState::HIGH;
```
again, don't know the exact variant names